spec/models/user_spec.rb

```diff
@@ -0,0 +1,21 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  describe 'validations' do
+    it { is_expected.to validate_presence_of(:name).allow_nil }
+    it { is_expected.to validate_length_of(:name).is_at_most(50) }
+    
+    it { is_expected.to validate_presence_of(:email).allow_nil }
+    it { is_expected.to validate_length_of(:email).is_at_most(100) }
+    
+    it { is_expected.to allow_value('test@test.com').for(:email) }
+    it { is_expected.not_to allow_value('abcdefg').for(:email) }     
+    
+    it { is_expected.to have_secure_password }
+    
+    context 'email uniqueness validation' do
+      before { create(:user) }
```

Comment:
これ必要ですか？

Reply:
`create(:user)`がない場合，次のようなエラーがでます

```
1) User validations email is expected to validate that :email is case-sensitively unique
     Failure/Error: it { is_expected.to validate_uniqueness_of(:email) }
     Shoulda::Matchers::ActiveRecord::ValidateUniquenessOfMatcher::ExistingRecordInvalid:
       validate_uniqueness_of works by matching a new record against an
       existing record. If there is no existing record, it will create one
       using the record you provide.
       While doing this, the following error was raised:
```

shoulda-matchersがすでにあるレコードを用いてuniqueかどうかを見てるようで，レコードがない場合，テーブルの定義からレコードを作るっぽいんですよね．
テーブルの定義から作る場合，name, emailにデフォルト値がないのでレコードが作れなくてエラーがでていました．

なので，beforeで一つuserを作っているというふうになっています．

Reply:
なるほど、勉強になりました
https://github.com/thoughtbot/shoulda-matchers/blob/0011a3bd9a6b07f8234954d0418a74bdeb393db7/lib/shoulda/matchers/active_record/validate_uniqueness_of_matcher.rb#L58